src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -0,0 +1,57 @@
+package nextstep.subway.application;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.LineRepository;
+import nextstep.subway.dto.LineRequest;
+import nextstep.subway.dto.LineResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+    private final LineRepository lineRepository;
+
+    public LineService(LineRepository lineRepository) {
+        this.lineRepository = lineRepository;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        Line persistLine = lineRepository.save(lineRequest.toLine());
+        return LineResponse.of(persistLine);
+    }
+
+    public List<LineResponse> findAllLines() {
+        List<Line> lines = lineRepository.findAll();
+
+        return lines.stream()
+            .map(LineResponse::of)
+            .collect(Collectors.toList());
+    }
+
+    public LineResponse findById(Long id) {
+        return lineRepository.findById(id).map(LineResponse::of)
+            .orElse(LineResponse.EMPTY);
+    }
+
+    @Transactional
+    public LineResponse updateLine(Long id, LineRequest lineRequest) {
+        Optional<Line> lineOptional = lineRepository.findById(id);
+        if (!lineOptional.isPresent()) {
+            return LineResponse.EMPTY;
+        }
+        Line line = lineOptional.get();
+        line.setName(lineRequest.getName());
```

Comment:
```suggestion
        line.update(...) 
```
동일한 기능이지만, `set***`의 형태는 저희교육과정에서 지양하고있어요! update***등의 의미있는 메서드명으로 사용해보면 어떨까요~?

Reply:
수정했습니다~! 

`updateLine` 메소드의 파라미터로 `name`과 `color`를 그대로 받았습니다. 파라미터 개수가 많이 생길 수도 있지만, `domain`이 `dto` 패키지에 의존성이 생기는 게 좋아보이지 않아 이렇게 구현했습니다.